### ISSUE TITLE
Document changes from v4 to v5

### ISSUE THREAD

**Issue by GregOriol** [pheanstalk/pheanstalk#272]
Currently, there is a section in the Readme about v4 but not about what has changed in v5: this makes it difficult for developers to migrate since the library has a few breaking changes... those should be explained.

**Comment by SamMousa** [comment]
Are you up for documenting this?

Since v5 is very strongly typed, my approach was roughly:
1. Update to version 5
2. Run `phpstan`
3. Fix error, go back to 2.

**Comment by SamMousa** [comment]
Did you see this btw? https://github.com/pheanstalk/pheanstalk/blob/v5/doc/library-design.md

**Comment by GregOriol** [comment]
@SamMousa I could only "document" what I guessed, I'm not a core developer of the library, I don't know what has actually changed and what is the best way to handle the changes... It's not a good practice to let users of library guess those things

**Comment by SamMousa** [comment]
I fully agree; unfortunately there's not a lot of maintainers (meaning its mostly just me) and maintain a lot of stuff and have other non maintenance related responsibilities as well.

**Comment by SamMousa** [comment]
Fixed in 5.0.7

**Comment by GregOriol** [comment]
Note that there could be other breaking changes that I don't know of...
